enumerable_methods.rb

```diff
@@ -0,0 +1,305 @@
+        return false unless self[i].is_a? arg
+      end
+    elsif arg.is_a? Regexp
+      (0...(length)).each do |i|
+        return false unless self[i].match(arg)
+      end
+    elsif !arg.is_a? Regexp and !arg.is_a? Class and !block_given?
+      (0...(length)).each do |i|
+        return false if self[i] != arg
+      end
+    else
+      (0...(length)).each do |i|
+        puts 'it entered here'
+        return false unless yield(self[i])
+      end
+    end
+    true
+  end
+
+  # 5th method
+
+  def my_any?(arg = nil)
+    return false if empty?
+
+    if arg.nil? and !block_given?
+      (0...(length)).each do |i|
+        return true if self[i] == true
+      end
+    elsif arg.is_a? Class
+      (0...(length)).each do |i|
+        return true if self[i].is_a? arg
+      end
+    elsif arg.is_a? Regexp
+      (0...(length)).each do |i|
+        return true if self[i].match(arg)
+      end
+    elsif !arg.is_a? Regexp and !arg.is_a? Class and !block_given?
+      (0...(length)).each do |i|
+        return true if self[i] == arg
+      end
+    else
+      (0...(length)).each do |i|
+        return true if yield(self[i])
+      end
+    end
+    false
+  end
+
+  # 6th method
+  def my_none?(arg = nil)
+    return true if empty?
+
+    if arg.nil? and !block_given?
+      (0...(length)).each do |i|
+        return false if self[i] == true
+      end
+    elsif arg.is_a? Class
+      (0...(length)).each do |i|
+        return false if self[i].is_a? arg
+      end
+    elsif arg.is_a? Regexp
+      (0...(length)).each do |i|
+        return false if self[i].match(arg)
+      end
+    elsif !arg.is_a? Regexp and !arg.is_a? Class and !block_given?
+      (0...(length)).each do |i|
+        return false unless self[i] != arg
+      end
+    else
+      (0...(length)).each do |i|
+        return false if yield(self[i])
+      end
+    end
+    true
+  end
+
+  # 7th method
+  def my_count(compare = nil)
+    to_array = Array(self)
+    return puts to_array.length if !block_given? && compare.nil?
+
+    count = 0
+
+    if compare.nil?
+      (0...(to_array.length)).each do |i|
+        count += 1 if yield(to_array[i])
+      end
+    elsif block_given? && !compare.nil?
+      puts 'warning: given block not used'
+      (0...(to_array.length)).each do |i|
+        count += 1 if compare == to_array[i]
+      end
+    else
+      (0...(to_array.length)).each do |i|
+        count += 1 if compare == to_array[i]
+      end
+    end
+    count
+  end
+
+  # This code was left as a comment on purpose
+  # so you can see the progression of the my_map method
+
+  # 8th method
+  #   def my_map
+  #     unless block_given?
+  #       return puts 'Enumerator #{self}:my_map'
+  #     else
+  #       new_arr = []
+  #       if self.is_a? Array
+  #         for i in 0...(self.length)
+  #             new_arr.push(yield(self[i]))
+  #         end
+  #       elsif self.is_a? Hash
+  #         key = self.keys
+  #         value = self.values
```

Comment:
Thank you for being thoughtful :pray: 
You don't have to do that unless it is specified, let's delete this commented out code from our project :blush: